src/Core/View.js

```diff
@@ -301,7 +302,22 @@ View.prototype.addLayer = function addLayer(layer, parentLayer) {
     }
     layer = _preprocessLayer(this, layer, provider);
     if (parentLayer) {
-        parentLayer.attach(layer);
+        if (layer.type == 'color') {
+            const colorLayers = this.getLayers(l => l.type === 'color');
+            colorLayers.push(layer);
+
+            const sumColorLayers = parentLayer.countTextureSamplersForColorLayers ?
+                parentLayer.countTextureSamplersForColorLayers(colorLayers) :
+                colorLayers.length;
+
+            if (sumColorLayers <= getMaxTextureSamplerCount()) {
+                parentLayer.attach(layer);
+            } else {
+                throw new Error('Cant add color layer: the maximum layer is reached');
```

Comment:
This is causing an error in `index.html` with the `menuGlobe` not being completed. Please adapt the example before merging.

Reply:
Ok PR Updated